wikimedia/wikipedia-ios
# Deep link into On This Day

**Fixes Phabricator ticket:**  https://phabricator.wikimedia.org/T172759

### Notes
* This is needed for opening the On This Day Widget (https://phabricator.wikimedia.org/T259840).
* @carolynlimadeo - Please confirm these views are what we want. (She and I went over the details previously, but I want to confirm this is as expected.) Can make tweaks in this PR (or later on).
* We're now opening this in a navigation stack, and so the header changes (and we no longer have an ending "back to Explore Feed" button).
* This only grabs the deep link for English Wikipedia's On This Day (https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today). It doesn't grab the pages that feed into On This Day (https://en.wikipedia.org/wiki/Wikipedia:Selected_anniversaries/August_31), as those are editable and perhaps the user wants to edit them. Additionally, it doesn't deep link for On This Day in non-English languages. This works for now - as we need to finish this up in order for the widget to work - but I've made a separate Phab ticket regarding these simplifications (and assigned to Josh).

### Test Steps
1. Test on iPad and iPhone.
1. Run app. Open On This Day card from Explore feed. Confirm it looks normal (especially with the header and button on bottom). Make sure you can open an article.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today'`. This should open an On This Day VC. Confirm the header looks like the mock ups below, and there is no "back to explore feed" at the bottom. Make sure you can open an article.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?5'`. Confirm that this opens the VC to the sixth event.
1. In a terminal window, run `xcrun simctl openurl booted 'https://en.wikipedia.org/wiki/Wikipedia:On_this_day/Today?afw'`. Confirm this opens and doesn't scroll. Garbage argument shouldn't crash the app or anything, it should just silently fail.

### Screenshots
<img width="397" alt="Screenshot 29" src="https://user-images.githubusercontent.com/9295855/92068527-04b58300-ed5c-11ea-98b0-8451b96b36a8.png">
